R/spliceHelperFunctions.R

```diff
@@ -96,232 +96,180 @@ findSpliceOverlapsByDist <-function(query, subject, ignore.strand=FALSE, maxDist
-  
-  startEndTable <- startEndTable %>% select(queryId,
-                                            subjectId, 
-                                            strand,
-                                            alternativeFirstExon,
-                                            alternativeLastExon,
-                                            alternativeTSS,
-                                            alternativeTES)
-  
-
-  ## annotate internal start and end first exons ##
-  subjectList <- unlist(subjectFullRng)
-  queryList <- unlist(queryFullRng)
-  sSpRngList <- unlist(sSpRng)
-  qSpRngList <- unlist(qSpRng)
-  
-  queryStart.subjectFull.Rng <- rep(unlist(queryStartRng),
-                                    elementNROWS(subjectFullRng))
-  queryStart.subjectFull.IdMap <- rep(1:length(queryStartRng),
-                                      elementNROWS(subjectFullRng))
-  
-  subjectStart.queryFull.Rng <- rep(unlist(subjectStartRng),
-                                    elementNROWS(queryFullRng))
-  subjectStart.queryFull.IdMap <- rep(1:length(subjectStartRng),
-                                      elementNROWS(queryFullRng))
-  
-  queryEnd.subjectFull.Rng <- rep(unlist(queryEndRng),
-                                  elementNROWS(subjectFullRng))
-  queryEnd.subjectFull.IdMap <- rep(1:length(queryEndRng),
-                                    elementNROWS(subjectFullRng))
-  
-  subjectEnd.queryFull.Rng <- rep(unlist(subjectEndRng),
-                                  elementNROWS(queryFullRng))
-  subjectEnd.queryFull.IdMap <- rep(1:length(subjectEndRng),
-                                    elementNROWS(queryFullRng))
-  
+  annotatedTable <- tibble(queryId = names(query),
+                           subjectId = names(subject),
+                           strand = as.character(unlist(unique(strand(query)))))
+  ###############################PRE-PROCESSING#################################
   #for intron retention/exon skipping
-  querySplice.subjectFullQuerySplice.Rng <- rep(unlist(qSpRng, use.names=F),
-                                     rep(elementNROWS(subjectFullRng),
-                                         times=elementNROWS(qSpRng)))
-  subjectFullQuerySplice.Rng <- unlist(rep(subjectFullRng,
-                                                       times= elementNROWS(qSpRng)), use.names=F)
-  querySplice.subjectFullQuerySplice.IdMap <- rep(1:length(qSpRng),elementNROWS(qSpRng)*elementNROWS(subjectFullRng))
-  
-  subjectSplice.queryFullsubjectSplice.Rng <- rep(unlist(sSpRng, use.names=F),rep(elementNROWS(queryFullRng), times=elementNROWS(sSpRng)))
-  queryFullSubjectSplice.Rng <- unlist(rep(queryFullRng, times= elementNROWS(sSpRng)), use.names=F)
-  subjectSplice.queryFullSubjectSplice.IdMap <- rep(1:length(sSpRng), elementNROWS(sSpRng)*elementNROWS(queryFullRng))
-  
+  querySplice.subjectFullQuerySplice.Rng <- expandMatchedRanges(querySpliceRng, subjectFullRng)
```

Comment:
the expandMatchedRanges seems very fast. We can move the creation of all of these objects into the sub-functions. While we will calculated the same object multiple times, we have to store less objects at any time, and it improved readability of the main function. You could compare the running time when you do this, if this is only marginally slower than it might be a good idea.

Reply:
The expandMatchedRanges() are now called from sub-functions.